scripts/json2synapse.py

```diff
@@ -0,0 +1,217 @@
+neuro = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/neuro.json'
+neurofibromatosis = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/neurofibromatosis.json'
+ngs = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/ngs.json'
+tool = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/tool.json'
+toolExtended = 'https://raw.githubusercontent.com/Sage-Bionetworks/synapseAnnotations/master/synapseAnnotations/data/toolExtended.json'
+
+paths = [analysis, sageCommunity, array, cancer, dhart, experimentalData, genie, neuro, neurofibromatosis, ngs, tool, toolExtended]
+names = ['analysis', 'sageCommunity', 'array', 'cancer', 'dhart', 'experimentalData', 'genie', 'neuro', 'neurofibromatosis', 'ngs', 'tool', 'toolExtended']
+
+
+all_modules = []
+
+tableSynId = "syn10242922"
+key = ["key", "value"]
+
+
+def json2flatten(path, module):
+    # fetch and read raw json objects from its' github url and decode the json object to its raw format
+    json_record = pandas.read_json(path)
+
+    # grab annotations with empty enumValue lists i.e. don't require normalization and structure their schema
+    empty_vals = json_record.loc[json_record.enumValues.str.len() == 0]
+    empty_vals = empty_vals.drop('enumValues', axis=1)
+
+    empty_vals['enumValues_description'] = ""
+    empty_vals['enumValues_source'] = ""
+    empty_vals['enumValues_value'] = ""
+    empty_vals['module'] = module
+
+    empty_vals.set_index(empty_vals['name'], inplace=True)
+
+    # for each value list object
+    flatten_vals = []
+    json_record = json_record.loc[json_record.enumValues.str.len() > 0]
+    json_record.reset_index(inplace=True)
+
+    for i, jsn in enumerate(json_record['enumValues']):
+        df = pandas.io.json.json_normalize(json_record['enumValues'][i])
+        # re-name columns to match table on synapse schema
+        df = df.rename(columns={'value': 'enumValues_value', 'description': 'enumValues_description',
+                                'source': 'enumValues_source'})
+
+        # grab key information in its row, expand it by values dimention and append its key-columns to flattened values
+        rows = json_record.loc[i:i, json_record.columns != 'enumValues']
+        repeats = pandas.concat([rows] * len(df.index))
+        repeats.set_index(df.index, inplace=True)
+        flatten_df = pandas.concat([repeats, df], axis=1)
+        # append module category / annotating the annotations for module filtering
+        flatten_df['module'] = module
+        flatten_df.set_index(flatten_df['name'], inplace=True)
+
+        flatten_vals.append(flatten_df)
+
+    flatten_vals.append(empty_vals)
+
+    return flatten_vals
+
+
+def rightJoin(df1, df2, key, suffixes):
```

Comment:
By right join I'm identifying keys ['key', 'value', 'module'] that have been changed. 
here is a toy example of what is generated by the rightJoin() function.
```python 
In []: dd = {'col1': [1,1,1], 'col2': [2,2,2], 'key1':['a', 'b', 'c'], 'key2':['x', 'y', 'z'], 'key3':['(', '*',')']}
     ...: df1 = pandas.DataFrame(data= dd, index=['key1', 'key2', 'key3'])
     ...: df1.set_index(keys=[df1.key1, df1.key2, df1.key3])
     ...: df1 = df1.set_index(keys=[df1.key1, df1.key2, df1.key3])
     ...: dd = {'col1': [1,1,1], 'col2': [3,2,3], 'key1':['a', 'b', 'change me'], 'key2':['x', 'y', 'z'], 'key3':['(', '*',')']}
     ...: df2 = pandas.DataFrame(data= dd)
     ...: df2.set_index(keys=[df2.key1, df2.key2, df2.key3])
     ...: rightJoin(df1=df1, df2=df2, key=['key1', 'key2', 'key3'], suffixes=('_cur', '_new'))
     ...:
Out[]:
   col1_cur  col2_cur       key1 key2 key3  col1_new  col2_new  Exist
0       1.0       2.0          a    x    (         1         3   True
1       1.0       2.0          b    y    *         1         2   True
2       NaN       NaN  change me    z    )         1         3  False
```